### ISSUE TITLE
[Bug]: Twitter announce incorrect URL

### ISSUE THREAD

**Issue by pgundlach** [goreleaser/goreleaser#2622]
### What happened?

I use twitter announce (which works fine), but it posted the incorrect URL. 

### How can we reproduce this?

This is most likely not a bug in Goreleaser and probably not released to Twitter announce, but to the value of `.GitURL`.

When importing a new repsitory on Github, it suggest to add a remote like: 

    git remote add origin https://github.com/speedata/ets.git

(notice the `.git` suffix)

Now I set up Twitter announce with the following setting:

```
announce:
  twitter:
    # Whether its enabled or not.
    # Defaults to false.
    enabled: true
```

and on `goreleaser release --rm-dist` it publishes a twitter message with an incorrect link. I don't have it anymore but I strongly believe it was `https://github.com/speedata/ets.git/releases/tag/v0.0.2`, including the `.git` suffix. This link gives a 404.

Perhaps it is possible to strip the `.git` suffix from the `.GitURL` or to override this in the yaml file? The `github_urls:` are documented to be for Github enterprise only, so I didn't try them.

Sorry that I have deleted the link too quickly. 

### goreleaser version

```bash
goreleaser --version
goreleaser version 0.183.0
commit: be6199f08167cc85b0b16cefc02d8106237853b1
built by: homebrew

https://goreleaser.com
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by developer-guy** [comment]
You can assign it to us @pgundlach if you don't mind @caarlos0? Because we found the code that caused that issue, we also noticed that there were other bugs. For example, the default template URL for GitLab is also wrong.

**Comment by caarlos0** [comment]
oh nice, sure go ahead @developer-guy

**Comment by developer-guy** [comment]
We addressed some of the bugs in PR #2623. The other bug that we found is that we're using a `defaultMessageTemplate` variable store message of the announce, but this URL within this template is suitable for GitHub; in GitLab, the URL is a bit different, GitLab uses `/-/releases/v0.1.3.` for releases. So, we ensure that we are using the correct message template according to the git URL configured by the user. For example, if the user uses GitLab, we should prefer to use `defaultGitLabMessageTemplate.`

cc: @Dentrax @erkanzileli
